server/src/handle_chaos_message.rs

```diff
@@ -339,75 +389,65 @@ fn dispatch_pt_object(
 }
 
 fn make_severity(proto: &chaos_proto::chaos::Severity) -> Result<Severity, Error> {
-    let effect = if proto.has_effect() {
-        make_effect(proto.get_effect())
-    } else {
-        bail!("Severity has no effect");
-    };
-
-    let priority = if proto.has_priority() {
-        Some(proto.get_priority())
-    } else {
-        None
-    };
-
-    let color = if proto.has_color() {
-        Some(proto.get_color().to_string())
-    } else {
-        None
-    };
-
-    let wording = if proto.has_wording() {
-        Some(proto.get_wording().to_string())
-    } else {
-        None
-    };
+    let effect = proto
+        .effect
+        .as_ref()
+        .ok_or_else(|| format_err!("'Severity' has no 'effect'"))?
+        .enum_value()
+        .map_err(|value| format_err!("'{}' is not a valid 'Effect'", value))?;
+    let effect = make_effect(effect);
 
     let result = Severity {
-        wording,
-        color,
-        priority,
+        wording: proto.wording.clone(),
+        color: proto.color.clone(),
+        priority: proto.priority,
         effect,
     };
     Ok(result)
 }
 
-pub fn make_effect(proto: chaos_proto::gtfs_realtime::Alert_Effect) -> Effect {
-    use chaos_proto::gtfs_realtime::Alert_Effect;
+pub fn make_effect(proto: chaos_proto::gtfs_realtime::alert::Effect) -> Effect {
+    use chaos_proto::gtfs_realtime::alert;
     match proto {
-        Alert_Effect::NO_SERVICE => Effect::NoService,
-        Alert_Effect::UNKNOWN_EFFECT => Effect::UnknownEffect,
-        Alert_Effect::SIGNIFICANT_DELAYS => Effect::SignificantDelays,
-        Alert_Effect::MODIFIED_SERVICE => Effect::ModifiedService,
-        Alert_Effect::DETOUR => Effect::Detour,
-        Alert_Effect::REDUCED_SERVICE => Effect::ReducedService,
-        Alert_Effect::ADDITIONAL_SERVICE => Effect::AdditionalService,
-        Alert_Effect::OTHER_EFFECT => Effect::OtherEffect,
-        Alert_Effect::STOP_MOVED => Effect::StopMoved,
+        alert::Effect::NO_SERVICE => Effect::NoService,
+        alert::Effect::UNKNOWN_EFFECT => Effect::UnknownEffect,
+        alert::Effect::SIGNIFICANT_DELAYS => Effect::SignificantDelays,
+        alert::Effect::MODIFIED_SERVICE => Effect::ModifiedService,
+        alert::Effect::DETOUR => Effect::Detour,
+        alert::Effect::REDUCED_SERVICE => Effect::ReducedService,
+        alert::Effect::ADDITIONAL_SERVICE => Effect::AdditionalService,
+        alert::Effect::OTHER_EFFECT => Effect::OtherEffect,
+        alert::Effect::STOP_MOVED => Effect::StopMoved,
     }
 }
 
-fn make_cause(proto: &chaos_proto::chaos::Cause) -> Cause {
-    Cause {
-        wording: proto.get_wording().to_string(),
-        category: proto.get_category().get_name().to_string(),
-    }
+fn make_cause(proto: &chaos_proto::chaos::Cause) -> Result<Cause, Error> {
+    let wording = proto
+        .wording
+        .as_ref()
+        .ok_or_else(|| format_err!("'Cause' has no 'wording'"))?
+        .to_string();
```

Comment:
If you want to save a line : 

```suggestion
    let wording = proto
        .wording
        .clone()
        .ok_or_else(|| format_err!("'Cause' has no 'wording'"))?;
```